### PR TITLE
Move from chai to earljs

### PR DESCRIPTION
**Pull request by hasparus**

### CODE REVIEW

File: packages/backend/test/mock.ts [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
-  const clone = { ...overrides }
+import { Mock as MockFunction, mockFn } from 'earljs'
+
+export type MockedObject<T> = {
+  [P in keyof T]: T[P] extends (...args: any[]) => any
+    ? MockFunction.Of<T[P]>
+    : T[P]
+} & T
+
+export function mock<T>(overrides: Partial<T> = {}): MockedObject<T> {
```
Comment (hasparus) [inline packages/backend/test/mock.ts:9]: Let me know what you think about this.

This function will now turn `{}` overrides into a mock object where every method is a mock function made with Earl's `mockFn`.

File: packages/backend/test/mock.ts [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
-      if (value !== undefined) {
+      if (value && typeof value !== 'function') {
         return value
       }
-      const name = String(property)
-      return () => {
-        throw new Error(
-          `Cannot call .${name}() - no mock implementation provided.`
-        )
+      if (isMockFunction(value)) {
```
Comment (sz-piotr) [inline packages/backend/test/mock.ts:17]: If we did the transformation before creating the proxy we could keep the old logic inside the proxy.

File: packages/backend/test/mock.ts [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
+        return value
       }
+      const m = mockFn(value || (() => {}))
```
Comment (sz-piotr) [inline packages/backend/test/mock.ts:20]: `mock<Foo>({ x: 0 })` would create a mock with x being a `mockFn`

File: packages/backend/test/mock.ts [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
+      Reflect.set(target, property, m, receiver)
+      return m
     },
   })
-  return proxy as T
+
+  return proxy
+}
+
+function isMockFunction(x: unknown): x is MockFunction<any[], unknown> {
```
Comment (sz-piotr) [inline packages/backend/test/mock.ts:29]: Ideally this could be exported from earl

File: packages/backend/test/core/StatusService.test.ts [hunk 5]
```diff
@@ -9,7 +9,7 @@ describe('StatusService', () => {
       barService: new BarService(),
     })
 
-    expect(statusService.getStatus()).to.deep.eq({
+    expect<object>(statusService.getStatus()).toEqual({
```
Comment (sz-piotr) [inline packages/backend/test/core/StatusService.test.ts:12]: Use either `unknown` or `object` everywhere. I am not sure how i like the fact that this fails for infinite types.

**Comment by sz-piotr** [comment]
Let me know what you think of the changes to `mock` and we can merge

**Pull request merged**